### PR TITLE
[bazel] Split gtest and gmock and become compatible with Bazel incompatible changes

### PR DESCRIPTION
**Pull request by Yannic**
Fixes googletest for upcoming `--incompatible_load_cc_rules_from_bzl` (https://github.com/bazelbuild/bazel/issues/8743) and `--incompatible_load_python_rules_from_bzl` (https://github.com/bazelbuild/bazel/issues/9006).

This change also makes the build more hermetic.

### CODE REVIEW

File: googletest/src/gtest-port.cc [hunk 1]
```diff
@@ -83,7 +83,7 @@
 #include "gtest/gtest-message.h"
 #include "gtest/internal/gtest-internal.h"
 #include "gtest/internal/gtest-string.h"
-#include "src/gtest-internal-inl.h"
+#include "gtest-internal-inl.h"
```
Comment (gennadiycivil) [inline googletest/src/gtest-port.cc]: How is this related to the bazel change? Please revert and just leave the bazel related changes.

File: googletest/src/gtest-printers.cc [hunk 2]
```diff
@@ -48,7 +48,7 @@
 #include <ostream>  // NOLINT
 #include <string>
 #include "gtest/internal/gtest-port.h"
-#include "src/gtest-internal-inl.h"
+#include "gtest-internal-inl.h"
```
Comment (gennadiycivil) [inline googletest/src/gtest-printers.cc]: How is this related to the bazel change? Please revert and just leave the bazel related changes.

File: googletest/src/gtest-test-part.cc [hunk 3]
```diff
@@ -31,7 +31,7 @@
 // The Google C++ Testing and Mocking Framework (Google Test)
 
 #include "gtest/gtest-test-part.h"
-#include "src/gtest-internal-inl.h"
+#include "gtest-internal-inl.h"
```
Comment (gennadiycivil) [inline googletest/src/gtest-test-part.cc]: How is this related to the bazel change? Please revert and just leave the bazel related changes.

File: googletest/src/gtest.cc [hunk 4]
```diff
@@ -119,7 +119,7 @@
 # include <sys/types.h>  // NOLINT
 #endif
 
-#include "src/gtest-internal-inl.h"
+#include "gtest-internal-inl.h"
```
Comment (gennadiycivil) [inline googletest/src/gtest.cc]: How is this related to the bazel change? Please revert and just leave the bazel related changes.

File: googletest/test/googletest-test2_test.cc [hunk 5]
```diff
@@ -32,7 +32,7 @@
 // Google Test work.
 
 #include "gtest/gtest.h"
-#include "googletest-param-test-test.h"
+#include "test/googletest-param-test-test.h"
```
Comment (gennadiycivil) [inline googletest/test/googletest-test2_test.cc]: How is this related to the bazel change? Please revert and just leave the bazel related changes.

File: googletest/test/production.cc [hunk 6]
```diff
@@ -30,6 +30,6 @@
 //
 // This is part of the unit test for gtest_prod.h.
 
-#include "production.h"
+#include "test/production.h"
```
Comment (gennadiycivil) [inline googletest/test/production.cc]: How is this related to the bazel change? Please revert and just leave the bazel related changes.

File: googletest/test/gtest_prod_test.cc [hunk 7]
```diff
@@ -30,8 +30,8 @@
 //
 // Unit test for gtest_prod.h.
 
-#include "production.h"
 #include "gtest/gtest.h"
+#include "test/production.h"
```
Comment (gennadiycivil) [inline googletest/test/gtest_prod_test.cc]: How is this related to the bazel change? Please revert and just leave the bazel related changes.

**Comment by gennadiycivil** [comment]
@Yannic Thank you for your PR, please take a look at the comments. In general it is vastly preferred to only have 1 change in the PR

**Comment by gennadiycivil** [comment]
Yes please split them into a separate PR. The code structure is not that easy to change , because there are hidden transformations beween the internal code that has somewhat different structure and the OSS code here. We would have to see *proof* that the code the way it is has issues with new Bazel.

**Comment by Yannic** [comment]
I split the changes to C++ files off into #2377. The Bazel tests are expectedly failing now. I verified that they'll pass with #2377 merged.

**Comment by gennadiycivil** [comment]
We should make this work without code structure alterations.  

Bazel changes 
--incompatible_load_cc_rules_from_bzl (bazelbuild/bazel#8743) and --incompatible_load_python_rules_from_bzl (bazelbuild/bazel#9006).

do not require that the project code change. They only ask to load rules from a different place.

**Comment by Yannic** [comment]
Okay, let's split this PR again to get a "load rules_{cc,python}" and a "split gtest and gmock" change then, ack?

**Comment by gennadiycivil** [comment]
> Okay, let's split this PR again to get a "load rules_{cc,python}" and a "split gtest and gmock" change then, ack?

I think the  "load rules_{cc,python}" is definitely valuable. 

The second one you would have to prove that it is indeed an improvement - a strong supporting case would have to be built. 

Thank you for this!

**Comment by Yannic** [comment]
I'll do some more investigations regarding the split into gtest and gmock.

Closing for now.
